controllers/users/checkHumanIdExchangeToken.js

```diff
@@ -40,6 +40,13 @@ const checkHumanIdExchangeToken = async (req, res) => {
       });
 
     if (user?.is_banned) return ErrorResponse.e400(res, 'User has banned by admin');
+    if (user?.verified_status === 'UNVERIFIED') {
+      return SuccessResponse(res, {
+        code: 500,
+        data: false,
+        message: 'User is UNVERIFIED'
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The HTTP status code 500 is typically used to indicate a server error. However, in this case, the user's verification status being 'UNVERIFIED' does not seem to be a server error but rather a client error or a normal application state. It would be more appropriate to use a different status code, such as 403 (Forbidden) or 412 (Precondition Failed), or even a 200 with an appropriate message if it's a valid state.

Also, you are using `SuccessResponse` to return an error condition which can be misleading. Consider creating a new method for error responses or use an existing one that fits the situation.

```diff
-    if (user?.verified_status === 'UNVERIFIED') {
-      return SuccessResponse(res, {
-        code: 500,
-        data: false,
-        message: 'User is UNVERIFIED'
-      });
-    }
+    if (user?.verified_status === 'UNVERIFIED') {
+      return ErrorResponse.e403(res, 'User is UNVERIFIED');
+    }
```
This change will make the response more semantically correct and easier to understand for clients of your API.


<!-- This is an auto-generated comment by OSS CodeRabbit -->